src/engine/QueryPlanner.cpp

```diff
@@ -238,52 +249,69 @@ QueryExecutionTree QueryPlanner::createExecutionTree(ParsedQuery& pq) const {
   SubtreePlan final = patternPlans[0];
 
   if (usePatternTrick) {
-    // Determine the column containing the subjects for which we are interested
-    // in their predicates.
-    auto it =
-        final._qet.get()->getVariableColumnMap().find(patternTrickTriple._s);
-    if (it == final._qet.get()->getVariableColumnMap().end()) {
-      AD_THROW(ad_semsearch::Exception::BAD_QUERY,
-               "The root operation of the"
-               "query excecution tree does"
-               "not contain a column for"
-               "variable " +
-                   patternTrickTriple._s +
-                   " required by the pattern"
-                   "trick.");
-    }
-    size_t subjectColumn = it->second;
-
-    bool isSorted =
-        final._qet->getRootOperation()->resultSortedOn() == subjectColumn;
-    // a and b need to be ordered properly first
-    vector<pair<size_t, bool>> sortIndices = {
-        std::make_pair(subjectColumn, false)};
+    if (final._qet->getRootOperation() != nullptr) {
+      // Determine the column containing the subjects for which we are
+      // interested in their predicates.
+      auto it =
+          final._qet.get()->getVariableColumnMap().find(patternTrickTriple._s);
+      if (it == final._qet.get()->getVariableColumnMap().end()) {
+        AD_THROW(ad_semsearch::Exception::BAD_QUERY,
+                 "The root operation of the "
+                 "query excecution tree does "
+                 "not contain a column for "
+                 "variable " +
+                     patternTrickTriple._s +
+                     " required by the pattern "
+                     "trick.");
+      }
+      size_t subjectColumn = it->second;
+      bool isSorted =
+          final._qet->getRootOperation()->resultSortedOn() == subjectColumn;
+      // a and b need to be ordered properly first
+      vector<pair<size_t, bool>> sortIndices = {
+          std::make_pair(subjectColumn, false)};
 
-    SubtreePlan orderByPlan(_qec);
-    std::shared_ptr<Operation> orderByOp(
-        new OrderBy(_qec, final._qet, sortIndices));
+      SubtreePlan orderByPlan(_qec);
+      std::shared_ptr<Operation> orderByOp(
+          new OrderBy(_qec, final._qet, sortIndices));
 
-    if (isSorted) {
-      orderByPlan._qet->setVariableColumns(final._qet->getVariableColumnMap());
-      orderByPlan._qet->setOperation(QueryExecutionTree::ORDER_BY, orderByOp);
+      if (isSorted) {
+        orderByPlan._qet->setVariableColumns(
+            final._qet->getVariableColumnMap());
+        orderByPlan._qet->setOperation(QueryExecutionTree::ORDER_BY, orderByOp);
+      }
+      SubtreePlan patternTrickPlan(_qec);
+      std::shared_ptr<Operation> countPred(new CountAvailablePredicates(
```

Comment:
`make_shared`

Reply:
Ok these are really everywhere so maybe consistency is more important than getting rid of the explicit `new`s